src/haproxy-sidecar/balena.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+set -ea
+
+[[ $VERBOSE =~ on|On|Yes|yes|true|True ]] && set -x
+
+which curl || apk add curl --no-cache
+which jq || apk add jq --no-cache
+
+network="${BALENA_APP_ID}_default"
+
+# shellcheck disable=SC2153
+for alias in ${ALIASES//,/ }; do
+    hostname="${alias}.${DNS_TLD}"
+    aliases="--alias ${hostname} ${aliases}"
+done
+
+while true; do
+    while [ "$(curl --silent --retry 3 --fail \
+      "${BALENA_SUPERVISOR_ADDRESS}/v1/device?apikey=${BALENA_SUPERVISOR_API_KEY}" \
+      -H "Content-Type:application/json" | jq -r '.update_pending')" = 'true' ]; do
+        sleep "$(( (RANDOM % 3) + 3 ))s"
+    done
```

Comment:
What's the purpose of this?

It'll also only work on balenaOS.

Reply:
Wait fir supervisor to finish syncing target state.